### PR TITLE
[New resource] - `azurerm_machine_learning_datastore_fileshare` Support Machine learning datastore fileshare

### PR DESCRIPTION
**Pull request by xuzhang3**
This PR depends on #19909 
```log
=== PAUSE TestAccMachineLearningDataStoreFileShare_Update
=== RUN   TestAccMachineLearningDataStoreFileShare_requiresImport
=== PAUSE TestAccMachineLearningDataStoreFileShare_requiresImport
=== CONT  TestAccMachineLearningDataStoreFileShare_accountKey
=== CONT  TestAccMachineLearningDataStoreFileShare_Update
=== CONT  TestAccMachineLearningDataStoreFileShare_sasToken
=== CONT  TestAccMachineLearningDataStoreFileShare_requiresImport
--- PASS: TestAccMachineLearningDataStoreFileShare_accountKey (658.60s)
--- PASS: TestAccMachineLearningDataStoreFileShare_sasToken (659.03s)
--- PASS: TestAccMachineLearningDataStoreFileShare_Update (865.18s)
--- PASS: TestAccMachineLearningDataStoreFileShare_requiresImport (967.17s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/machinelearning       995.634s

```

### CODE REVIEW

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for splitting these out @xuzhang3, however these also need to be converted to [typed resources as well](https://github.com/hashicorp/terraform-provider-azurerm/pull/19867#pullrequestreview-1237116730).

**Comment by xuzhang3** [comment]
@stephybun Thanks for reviewing this PR. Refactored into typed resource.

File: website/docs/r/machine_learning_datastore_fileshare.html.markdown [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+---
+* `account_key` - (Optional) The access key of the Storage Account. Conflicts with `shared_access_signature`.
+
+* `shared_access_signature` - (Optional) The Shared Access Signature of the Storage Account. Conflicts with `account_key`.
+
+* `description` - (Optional) Text used to describe the asset. Changing this forces a new Machine Learning DataStore to be created.
+
+* `is_default` - (Optional) Specifies whether this Machines Learning DataStore is the default for the Workspace. Defaults to `false`.
+
+~> **Note:** `is_default` can only be set to `true` on update.
```
Comment (katbyte) [inline website/docs/r/machine_learning_datastore_fileshare.html.markdown:96]: As in you cannot set it on creation? could we not just make an update call at the end of the create function to set this to true?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: website/docs/r/machine_learning_datastore_fileshare.html.markdown (hunk 1 above)
Comment (xuzhang3) [reply website/docs/r/machine_learning_datastore_fileshare.html.markdown:96]: This property seems managed by a separate API and not documented in the swagger. For now this property cannot updated. Checking with service team.

Comment (katbyte) [reply website/docs/r/machine_learning_datastore_fileshare.html.markdown:96]: if its managed by the service then "`is_default` can only be set to `true` on update." is not valid?

and then maybe we should just make it computed?

Comment (xuzhang3) [reply website/docs/r/machine_learning_datastore_fileshare.html.markdown]: Have changed `is_default`  to computed

**Review by katbyte — APPROVED** [review]
Thanks @xuzhang3 LGTM ☔

**Pull request merged**